File: nexus/poc/ui/src/api/sql.ts
// SQL playground: run DataFusion SQL over inline JSON rows.

import { postJson } from "./client";

export interface SqlResponse {
  ok: boolean;
  error: string | null;
  row_count: number;
  rows: Record<string, unknown>[];
}

export function runSql(query: string, rows: unknown[]): Promise<SqlResponse> {
  return postJson<SqlResponse>("/api/sql/query", { query, rows });
}

File: nexus/poc/backend/src/dto/sql.rs
//! Wire types for the SQL playground endpoint.

use serde::{Deserialize, Serialize};
use serde_json::Value;

/// Run `query` (DataFusion SQL) over an inline array of JSON rows.
#[derive(Debug, Clone, Deserialize)]
pub struct SqlRequest {
    pub query: String,
    /// The dataset, as a JSON array of objects. Defaults to empty.
    #[serde(default)]
    pub rows: Vec<Value>,
}

#[derive(Debug, Serialize)]
pub struct SqlResponse {
    pub ok: bool,
    pub error: Option<String>,
    pub row_count: usize,
    pub rows: Vec<Value>,
}
